sound/soc/intel/boards/bytcht_cx2072x.c

```diff
@@ -272,15 +272,16 @@ static int snd_byt_cht_cx2072x_probe(struct platform_device *pdev)
 		byt_cht_cx2072x_card.driver_name = DRIVER_NAME;
 	}
 
+	/* set pm ops */
+	if (mach->mach_params.sof_parent_device)
+		pdev->dev.driver->pm = &snd_soc_pm_ops;
```

Comment:
I found one (!) more example of assigning driver PM ops at runtime in drivers/net/wireless/ti/wlcore/main.c so, presumably, this should work...

Reply:
yes it's a bit hairy but Ranjani and I didn't see any other way to do it.